src/dali/plugin/input/csv/binance_com.py

```diff
@@ -0,0 +1,175 @@
+                        exchange=self.__BINANCE_COM,
+                        holder=self.account_holder,
+                        transaction_type=Keyword.BUY.value,
+                        spot_price=Keyword.UNKNOWN.value,
+                        crypto_in=(line[self.__AUTO_BASE_AMOUNT_SYMBOL]).split()[0],
+                        notes=f"Autoinvest buy with funding from {line[self.__FUND_SOURCE]}",
+                    )
+                )
+
+                quote_asset_symbol: str = line[self.__AUTO_QUOTE_AMOUNT_SYMBOL].split()[1]
+                quote_asset_amount: str = line[self.__AUTO_QUOTE_AMOUNT_SYMBOL].split()[0]
+                crypto_fee: str = line[self.__AUTO_TRADING_FEE_SYMBOL].split()[0]
+                crypto_out_with_fee: RP2Decimal = RP2Decimal(quote_asset_amount) + RP2Decimal(crypto_fee)
+                result.append(
+                    OutTransaction(
+                        plugin=self.__BINANCE_COM_CSV,
+                        unique_id=Keyword.UNKNOWN.value,
+                        raw_data=raw_data,
+                        timestamp=f"{line[self.__TIMESTAMP_INDEX]} -00:00",
+                        asset=quote_asset_symbol,
+                        exchange=self.__BINANCE_COM,
+                        holder=self.account_holder,
+                        transaction_type=Keyword.SELL.value,
+                        spot_price=Keyword.UNKNOWN.value,
+                        crypto_out_no_fee=quote_asset_amount,
+                        crypto_out_with_fee=str(crypto_out_with_fee),
+                        crypto_fee=crypto_fee,
+                        notes=f"Autoinvest withdrawal for the purchase of {line[self.__AUTO_BASE_AMOUNT_SYMBOL]}",
+                    )
+                )
+
+            return result
+
+    def parse_betheth_file(self, file_path: str) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        with open(file_path, encoding="utf-8") as csv_file:
+            lines = reader(csv_file)
+
+            header = next(lines)
+            self.__logger.debug("Header: %s", header)
+            for line in lines:
+                raw_data: str = self.__DELIMITER.join(line)
+                self.__logger.debug("Transaction: %s", raw_data)
+
+                result.append(
+                    InTransaction(
+                        plugin=self.__BINANCE_COM_CSV,
+                        unique_id=Keyword.UNKNOWN.value,
+                        raw_data=raw_data,
+                        timestamp=f"{line[self.__TIMESTAMP_INDEX]} -00:00",
+                        asset="BETH",
+                        exchange=self.__BINANCE_COM,
+                        holder=self.account_holder,
+                        transaction_type=Keyword.BUY.value,
+                        spot_price=Keyword.UNKNOWN.value,
+                        crypto_in=line[self.__BETHETH_AMOUNT],
+                        notes="Conversion from BETH -> ETH",
```

Comment:
BETH -> ETH: these two transactions are: ETH out + BETH in. Should the string say: ETH -> BETH?

Reply:
uh, yeah. Oops. Good catch.